packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2313,18 +2383,74 @@ Hardhat Network's forking functionality only works with blocks from at least spu
 
     return { maxFeePerGas, maxPriorityFeePerGas };
   }
-}
 
-function getCommonForTracing(networkId: number, blockNumber: number): Common {
-  try {
-    const common = new Common({ chain: networkId });
+  private _selectHardfork(
+    blockNumber: BN,
+    defaultName = this._vm._common.hardfork() as HardforkName
+  ): HardforkName {
+    function assertSupported(n: HardforkName) {
+      if (!HARDHAT_NETWORK_SUPPORTED_HARDFORKS.includes(n)) {
+        throw new InternalError(`Unsupported hardfork ${n}`);
```

Comment:
This error should give more context about what's going on, like:

`Trying to run a call or transaction using a block whose hardfork is ${x}, but it's not supported by Hardhat Network`

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1928/commits/f6a5dbe1fe1aa1007192aa4e2cf31e1f1c7c5426